src/kernel/heap.zig

```diff
@@ -381,12 +388,12 @@ const FreeListAllocator = struct {
             }
 
             // If there is enough unused space to the right of this node then create a smaller node
-            if ((@sizeOf(Header) + header.size) - alignment_padding - real_size > @sizeOf(Header)) {
-                header.next_free = insertFreeHeader(@ptrToInt(header) + real_size + alignment_padding, header.size + @sizeOf(Header) - real_size - alignment_padding - @sizeOf(Header), header.next_free);
+            if (header.size - alignment_padding > real_size) {
```

Comment:
Nice. I had the extra `@sizeOf(Header)` in there before to avoid underflow, but that doesn't seem to be a problem.